libmamba/src/solv-cpp/queue.hpp

```diff
@@ -0,0 +1,146 @@
+     *
+     * This is used privately within libsolv, hence the direct use of ``Queue`` as an attribute
+     * (for performances).
+     */
+    class ObjQueue
+    {
+    public:
+        using value_type = ::Id;
+        using reference = value_type&;
+        using const_reference = value_type const&;
+        using pointer = ::Id*;
+        using const_pointer = ::Id const*;
+        using size_type = std::size_t;
+        using difference_type = std::ptrdiff_t;
+        using iterator = pointer;
+        using const_iterator = const_pointer;
+        using reverse_iterator = std::reverse_iterator<iterator>;
+        using const_reverse_iterator = std::reverse_iterator<const_iterator>;
+
+        ObjQueue();
+        template <typename InputIt>
+        ObjQueue(InputIt first, InputIt last);
+        ObjQueue(std::initializer_list<value_type> elems);
+        ObjQueue(ObjQueue&& other);
+        ObjQueue(ObjQueue const& other);
+
+        ~ObjQueue();
+
+        auto operator=(ObjQueue&& other) -> ObjQueue&;
+        auto operator=(ObjQueue const& other) -> ObjQueue&;
+
+        auto size() const -> size_type;
+        auto capacity() const -> size_type;
+        auto empty() const -> bool;
+
+        auto insert(const_iterator pos, value_type id) -> iterator;
+        template <typename InputIt>
+        auto insert(const_iterator pos, InputIt first, InputIt last) -> iterator;
+        void push_back(value_type id);
+        void push_back(value_type id1, value_type id2);
+        auto erase(const_iterator pos) -> iterator;
+        void reserve(size_type new_cap);
+        void clear();
+
+        auto front() -> reference;
+        auto front() const -> const_reference;
+        auto back() -> reference;
+        auto back() const -> const_reference;
+        auto operator[](size_type pos) -> reference;
+        auto operator[](size_type pos) const -> const_reference;
+        auto at(size_type pos) -> reference;
+        auto at(size_type pos) const -> const_reference;
+        auto begin() -> iterator;
+        auto begin() const -> const_iterator;
+        auto cbegin() const -> const_iterator;
+        auto end() -> iterator;
+        auto end() const -> const_iterator;
+        auto cend() const -> const_iterator;
+        auto rbegin() -> reverse_iterator;
+        auto rbegin() const -> const_reverse_iterator;
+        auto crbegin() const -> const_reverse_iterator;
+        auto rend() -> reverse_iterator;
+        auto rend() const -> const_reverse_iterator;
+        auto crend() const -> const_reverse_iterator;
+        auto data() -> pointer;
+        auto data() const -> const_pointer;
+
+        template <template <typename, typename...> class C>
+        auto as() -> C<value_type>;
+
+        auto get() -> ::Queue*;
+        auto get() const -> ::Queue const*;
+
+    private:
+        friend void swap(ObjQueue& a, ObjQueue& b) noexcept;
+
+        ::Queue m_queue = {};
+
+        ObjQueue(std::nullptr_t);
```

Comment:
First time I see this kind of hack/technique XD
Consider making this one `explicit` to avoid unintional conversions inside the cpp